### ISSUE TITLE
CLIP Tokenizer conflict with dataloader

### ISSUE THREAD

**Issue by giacomorebecchi** [giacomorebecchi/food-pricing#30]
Inside the CLIP model, multiprocessing in the tokenizer gets in conflict with the multiprocessing in the dataloader.
This is the warning obtained:
```
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
	- Avoid using `tokenizers` before the fork if possible
	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
```

**Comment by giacomorebecchi** [comment]
Note that this can be solved together with #29 by using the tokenizer inside the dataloader rather than during the training step, exactly like it's done in other models (e.g. BERT)

**Comment by giacomorebecchi** [comment]
Given time constraints, a temporary fix for this is setting in the .env file the following parameter:
```
TOKENIZERS_PARALLELISM=False
```

See an in-depth discussion about this in the following [post on StackOverflow](https://stackoverflow.com/a/72926996/18244126).

**Comment by giacomorebecchi** [comment]
Remember to document this as the default .env in the README.md file

**Comment by giacomorebecchi** [comment]
A better solution would be to create a separate instance of the tokenizer, use it on single elements of the dataset, and do the padding later by overwriting the `default_collate` function in the DataLoader. However, this seems to be very cumbersome given the minor advantage it would provide with.
